### PR TITLE
Prevent GraphQL apps from starting when they have a `@Query` annotation on a void method

### PR DESCRIPTION
**Pull request by andymc12**
Per the MP GraphQL spec, a query method should always have a return type, and putting a `@Query` annotation on a void method is considered a deployment error.

This change enforces that and provides a new error message for deployment errors.

It also tests that the GraphQL application does not start, and that the new error message is printed when a query method has a void return type.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# This file follows the WebSphere Message Guidelines.
+# For more information, visit: 
+# http://washome.austin.ibm.com/xwiki/bin/view/MessagesTeam/
+#
+
+#-----------------------------------------------------------------------------------------------------------------------------
+# Messages used by MP GraphQL Impl -  range is 0000-2999, plus 9999 for the emergency message
+#-----------------------------------------------------------------------------------------------------------------------------
+
+ERROR_GENERATING_SCHEMA_CWMGQ0001E=CWMGQ0001E: An error was detected that prevents the MicroProfile GraphQL application from starting in web module, {0}.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: In this sentence
`An error was detected that prevents the MicroProfile GraphQL application from starting in web module, {0}.`

Was the error detected in MicroProfile GraphQL or web module {0}? Or do we know?

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ERROR_GENERATING_SCHEMA_CWMGQ0001E.explanation=A severe error was detected while processing the MP GraphQL application in the specified module. This error prevents the application from starting successfully.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: This explanation really just restates the description. Is there any more information we can offer about the kind of error or what might have caused it?

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops (hunk 1 above)
Comment (andymc12) [reply dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: The error is most likely in the web module, though it is possible that a bug in MP GraphQL could cause an exception that would result in this message being printed. But the intention of this message is to indicate an error in the web module.

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops (hunk 2 above)
Comment (andymc12) [reply dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: Not really.  This error message is printed as a sort of "catch-all" for any error in the application (web module) that would prevent the app from starting.  There should be accompanying information in the logs that would indicate the source of the problem.

**Comment by andymc12** [comment]
#run-libby-bot #build

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops (hunk 1 above)
Comment (dmuelle) [reply dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: ok- in that case change to:

`An error in the {0} web module  prevented the MicroProfile GraphQL application from starting.`

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops (hunk 2 above)
Comment (dmuelle) [reply dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: ok

File: dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops (hunk 1 above)
Comment (andymc12) [reply dev/com.ibm.ws.microprofile.graphql.1.0/resources/com/ibm/ws/microprofile/graphql/resources/MPGraphQL.nlsprops]: That sounds better - thanks for the suggestion.  I've incorporated it in the latest commit.

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
#run-libby-bot

**Pull request merged**